tests/components/homepluscontrol/test_integration.py

```diff
@@ -0,0 +1,794 @@
+"""Test the Legrand Home+ Control integration."""
+import asyncio
+from unittest.mock import patch
+
+from homepluscontrol.homeplusplant import (
+    PLANT_TOPOLOGY_BASE_URL,
+    PLANT_TOPOLOGY_RESOURCE,
+)
+
+from homeassistant import config_entries, setup
+from homeassistant.components.homepluscontrol import api
+from homeassistant.components.homepluscontrol.const import DOMAIN, PLANT_URL
+
+
+async def entity_assertions(
+    hass,
+    num_exp_entities,
+    num_exp_devices=None,
+    coordinator=None,
+    expected_entities=None,
+    expected_devices=None,
+):
+    """Assert number of entities and devices."""
+    entity_reg = await hass.helpers.entity_registry.async_get_registry()
+    device_reg = await hass.helpers.device_registry.async_get_registry()
+
+    if coordinator is None:
+        coordinator = hass.data["homepluscontrol"][
+            "homepluscontrol_entry_id_coordinator"
+        ]
+
+    if num_exp_devices is None:
+        num_exp_devices = num_exp_entities
+
+    assert len(hass.data[DOMAIN]["entities"].keys()) == num_exp_entities
+    if coordinator.data is not None:
+        assert len(coordinator.data.keys()) == num_exp_entities
+    assert len(entity_reg.entities.keys()) == num_exp_entities
+    assert len(device_reg.devices.keys()) == num_exp_devices
+
+    if expected_entities is not None:
+        for exp_entity, present in expected_entities.items():
+            assert bool(entity_reg.async_get(exp_entity)) == present
+
+    if expected_devices is not None:
+        for exp_device, present in expected_devices.items():
+            assert bool(device_reg.async_get(exp_device)) == present
+
+
+async def test_loading(hass, mock_config_entry):
+    """Test component loading."""
+    await setup.async_setup_component(hass, "http", {})
+    assert hass.http.app
+    await setup.async_setup_component(hass, "homepluscontrol", {})
+
+    await mock_config_entry.async_setup(hass)
```

Comment:
Always use the core interfaces in the tests. Here use `hass.config_entries.async_setup`.

Reply:
Noted. This has been adjusted in the latest code now.